Clinical-Genomics/scout
# Pheno search api - Fix #2749

This PR adds a functionality or fixes a bug.

Add an HPO term search endpoint as 
/api/v1/phenotype/search/<QUERY_STRING>

As a bonus, adds CORS header support for js apps via Flask-CORS.

**How to test**:
1. test a search with an hpo term against the api, and notice hpo terms being returned, e.g. `http://localhost:5000/api/v1/phenotypes/search/kidney`
![Screenshot 2021-09-14 at 18 04 13](https://user-images.githubusercontent.com/758570/133293428-2a567f62-0450-4506-87ff-04685ea8cd55.png)
Also note that CORS sensitive parts of Scout are still working as before, e.g. IGV viewer with external tracks.

2. test with a scout-react install and note how CORS errors disappear when this branch is used.

Before:
![Screenshot 2021-09-14 at 16 40 22](https://user-images.githubusercontent.com/758570/133293580-cba2a5d3-5771-4fda-a94b-d3aeea58350f.png)

After:
![Screenshot 2021-09-14 at 17 20 20](https://user-images.githubusercontent.com/758570/133293468-5afb10c2-1872-4188-a72c-140587987fda.png)

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
